### PR TITLE
kode-mono font added

### PR DESCRIPTION
**Pull request by isaozler**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by enobayram** [comment]
@isaozler I'm getting the following failure when I try to `nix build` `kode-mono` from this branch:
```
$ nix build --no-link --print-out-paths github:/NixOS/nixpkgs?rev=b7e17bee1a6d08f2b357c8cf5f54a7fd7fd4af7f#kode-mono
error: builder for '/nix/store/85d242dwi87rhq8w01a5wkvqgf8dlr79-kode-mono-1.016.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking source archive /nix/store/dkxxp1ry1y3m9475zha7r3amvn94dz8w-source
       > source root is source
       > patching sources
       > configuring
       > no configure script, doing nothing
       > building
       > no Makefile or custom buildPhase, doing nothing
       > installing
       > install: missing file operand
       > Try 'install --help' for more information.
       For full logs, run 'nix log /nix/store/85d242dwi87rhq8w01a5wkvqgf8dlr79-kode-mono-1.016.drv'.
```

File: pkgs/data/fonts/kode-mono/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (drupol) [inline pkgs/data/fonts/kode-mono/default.nix:3]: Can you please use the `finalAttrs` pattern here?

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

File: pkgs/data/fonts/kode-mono/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "kode-mono";
+  version = "1.016";
+
+  src = fetchzip {
+    url = "https://github.com/isaozler/kode-mono/releases/download/${version}/kode-mono-fonts.zip";
+    sha256 = "sha256-GATMXmyBc8XuQVpv5YLPiI0v8tu+8j01yMNADYB6QIo=";
```
Comment (drupol) [inline pkgs/data/fonts/kode-mono/default.nix:9]: I think you should replace `sha256` with `hash`.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/kode-mono/default.nix (hunk 1 above)
Comment (isaozler) [reply pkgs/data/fonts/kode-mono/default.nix]: Thanks 👍

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Please squash your commits and it's good to go.

**Comment by isaozler** [comment]
Thanks for your quick reply @drupol 🙏

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
I've been too fast to hit the button. Damn.

Just fyi, the commit log has a specific structure and yours doesn't respect it. Have you read the contribution guidelines?

**Comment by isaozler** [comment]
> I've been too fast to hit the button. Damn.
> 
> Just fyi, the commit log has a specific structure and yours doesn't respect it. Have you read the contribution guidelines?

I did, but I was also too fast, squashing the commits. I will pay extra attention next time 👍 Thanks for reminding me! If you want me to change anything, please let me know.

**Comment by drupol** [comment]
Thank you very much!

Since this is already committed to master, there's nothing else to do.

**Pull request merged**